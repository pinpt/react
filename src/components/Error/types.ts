import { ReactElement, ReactNode } from 'react';
import { IActionLinkProps } from '../Internal/ActionLink';
import { ILogoProps } from '../Logo';

export interface IErrorProps {
	className?: string;
	logo?: ReactElement<ILogoProps | IActionLinkProps>;
	error?: string;
	title?: string;
	description?: string;
	link?: ReactNode;
}
