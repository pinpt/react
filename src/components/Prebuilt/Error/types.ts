import { ReactElement } from "react";
import { ISite } from "../../../lib";

export interface IPrebuiltErrorProps {
	renderLogo?: (site) => ReactElement;
	site: ISite;
};
