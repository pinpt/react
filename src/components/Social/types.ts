export interface ISocialProps {
	className?: string;
	newTab?: boolean;
	href?: string;
	onClick?: () => void;
}
