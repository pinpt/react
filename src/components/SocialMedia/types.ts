export interface ISocialMediaLinkProps {
	className?: string;
	newTab?: boolean;
	href?: string;
	onClick?: () => void;
}

export interface ISocialMediaShareProps {
	className?: string;
	newTab?: boolean;
	url: string;
	onClick?: () => void;
}
