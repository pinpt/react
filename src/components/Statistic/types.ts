export interface IStatisticProps {
	count: number;
	className?: string;
}
