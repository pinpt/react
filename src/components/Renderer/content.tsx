import mediumZoom from 'medium-zoom';
import { forwardRef, useEffect } from 'react';
import { slugifyContent } from '../../lib/string';
import { CoverMediaType } from '../../lib/types/content';
import { Document } from './';
import React from 'react';
import type { ICoverMedia } from '../../lib/types/content';

const ImageMedia = ({ src, title = '', zoomable = false }: { src: string; title?: string; zoomable?: boolean }) => {
	useEffect(() => {
		if (typeof window !== 'undefined') {
			mediumZoom('.medium-zoom-cover');
		}
	}, []);

	return (
		<div className="Pinpoint image">
			<img src={src} alt={title} className={`${zoomable ? 'medium-zoom-cover' : ''}`} />
		</div>
	);
};

const VideoMedia = ({ src }: { src: string }) => {
	return (
		<div className="Pinpoint video">
			<video controls src={src} />
		</div>
	);
};

const YoutubeMedia = ({
	id,
	metadata,
}: {
	id: string;
	metadata?: Record<string, any>;
}) => {
	const posterUrl = `https://i.ytimg.com/vi/${id}/${metadata?.poster ?? 'hqdefault'}.jpg`;
	return (
		<div className="Pinpoint youtube">
			<div
				className="yt"
				style={{ backgroundImage: `url("${posterUrl}")` }}
				data-url={`https://www.youtube-nocookie.com/embed/${encodeURIComponent(id)}?autoplay=1`}
			>
				<button className="play-button" />
			</div>
		</div>
	);
};

export const CoverMedia = ({
	media,
	title,
	zoomable,
}: {
	media?: ICoverMedia;
	title?: string;
	zoomable?: boolean;
}) => {
	if (!media) {
		return <></>;
	}
	let content: React.ReactNode = null;
	switch (media.type) {
		case CoverMediaType.None: {
			return <></>;
		}
		case CoverMediaType.Image: {
			content = <ImageMedia src={media.value || media.placeholderImage} title={title} zoomable={zoomable} />;
			break;
		}
		case CoverMediaType.Video: {
			content = <VideoMedia src={media.value} />;
			break;
		}
		case CoverMediaType.Youtube: {
			content = <YoutubeMedia id={media.value} metadata={media.metadata} />;
			break;
		}
	}
	return (
		<section className="Pinpoint cover covermedia">
			<div className="Pinpoint media-container">{content}</div>
		</section>
	);
};

interface ContentProps {
	id: string;
	title?: string;
	document?: any;
	coverMedia?: ICoverMedia;
	limit?: number;
	divider?: boolean;
}

const Content = forwardRef((props: ContentProps, ref: any) => {
	return (
		<article ref={ref}>
			{props.coverMedia && <CoverMedia media={props.coverMedia} title={props.title} />}
			<section className="Pinpoint content">
				<Document node={props.document} limit={props.limit} />
				{props.limit && props.document.content?.length > props.limit && (
					<div className="Pinpoint continue">
						<a href={slugifyContent(props.id, props.title)}>Continue Reading â†’</a>
					</div>
				)}
			</section>
			{props.divider && <hr className="Pinpoint divider" />}
		</article>
	);
});

export default Content;
