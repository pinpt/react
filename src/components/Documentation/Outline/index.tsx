import React from 'react';
import { useCallback, useState } from 'react';
import { getDocumentHeadings, IDocumentHeading } from '../../../lib';
import { IContent } from '../../../lib/types/content';
import { ISite } from '../../../lib/types/site';
import ActionLink from '../../Internal/ActionLink';

export interface IOutlineProps {
	className?: string;
	itemClassName?: string;
	site: ISite;
	entries?: IContent[];
	onClick?: (entry: IContent, anchor?: string) => void;
	active?: string;
	activeAnchor?: string;
}

export interface IOutlineEntryProps extends Omit<IOutlineProps, 'className' | 'entries' | 'active'> {
	entry: IContent;
	isActive: boolean;
	headings: IDocumentHeading[];
	isParentActive: boolean;
}

const Entry = (props: IOutlineEntryProps) => {
	const { itemClassName = '', onClick, activeAnchor = '', entry, isActive, headings, isParentActive } = props;
	const [collapsed, setCollapsed] = useState(true);
	const hasSubMenu = headings.length > 0;
	const isSubMenuOpen = hasSubMenu && !collapsed;

	const handleClickHeading = useCallback(() => {
		setCollapsed(false);
		onClick?.(entry);
	}, [entry, onClick]);

	return (
		<div className="Parent">
			<div className="Top">
				{hasSubMenu ? (
					<>
						{isSubMenuOpen ? (
							<svg
								onClick={() => setCollapsed((c) => !c)}
								aria-hidden="true"
								className="toggleControl"
								focusable="false"
								data-prefix="fas"
								data-icon="chevron-down"
								role="img"
								xmlns="http://www.w3.org/2000/svg"
								viewBox="0 0 448 512"
							>
								<path
									fill="currentColor"
									d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
								></path>
							</svg>
						) : (
							<svg
								onClick={() => setCollapsed((c) => !c)}
								aria-hidden="true"
								className="toggleControl"
								focusable="false"
								data-prefix="fas"
								data-icon="chevron-right"
								role="img"
								xmlns="http://www.w3.org/2000/svg"
								viewBox="0 0 320 512"
							>
								<path
									fill="currentColor"
									d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
								></path>
							</svg>
						)}
					</>
				) : (
					<div className="spacer" />
				)}
				<ActionLink
					className={`Pinpoint Documentation Outline Item ${isActive ? 'active' : ''} ${itemClassName}`}
					onClick={handleClickHeading}
				>
					{entry.title}
				</ActionLink>
			</div>
			{isSubMenuOpen && (
				<div className="SubMenu">
					{headings.map((h) => {
						const anchorActive = isParentActive && activeAnchor === h.id;
						return (
							<ActionLink
								key={h.id}
								onClick={onClick ? () => onClick(entry, h.id) : undefined}
								className={`SubItem ${anchorActive ? 'active' : ''}`}
							>
								{h.title}
							</ActionLink>
						);
					})}
				</div>
			)}
		</div>
	);
};

const Outline = (props: IOutlineProps) => {
	const { className = '', entries, active = '', activeAnchor = '' } = props;

	return (
		<div className={`Pinpoint Documentation Outline Wrapper ${className}`}>
			{entries?.map((entry) => {
				const isParentActive = active === entry.id;
				const isActive = isParentActive && !activeAnchor;
				const headings = getDocumentHeadings(entry.document, entry.title, [1]);
				return (
					<Entry
						key={`outline-${entry.id}`}
						{...props}
						isActive={isActive}
						isParentActive={isParentActive}
						headings={headings}
						entry={entry}
					/>
				);
			})}
		</div>
	);
};

export default Outline;
