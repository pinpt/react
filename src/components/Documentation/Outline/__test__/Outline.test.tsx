import renderer from 'react-test-renderer';
import Outline from '..';
import { IContent } from '../../../../lib/types/content';
import entries from '../../../Prebuilt/__data__/testEntries.json';
import site from '../../../Prebuilt/__data__/testSite.json';

test('Test default state', () => {
	const component = renderer.create(<Outline site={site} entries={entries as IContent[]} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test with click handler', () => {
	const component = renderer.create(
		<Outline site={site} entries={entries as IContent[]} onClick={(entry, hash) => console.log(entry.title, hash)} />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test with active', () => {
	const component = renderer.create(
		<Outline
			site={site}
			entries={entries as IContent[]}
			onClick={(entry) => console.log(entry.title)}
			active="Cq4Ong4PJwMYygleMjvk"
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test with active', () => {
	const component = renderer.create(
		<Outline
			site={site}
			entries={entries as IContent[]}
			onClick={(entry) => console.log(entry.title)}
			active="Cq4Ong4PJwMYygleMjvk"
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
