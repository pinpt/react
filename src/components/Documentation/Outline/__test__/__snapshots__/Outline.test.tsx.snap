// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test default state 1`] = `
<div
  className="Pinpoint Documentation Outline Wrapper "
>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Linked Content Blocks
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Publishing Controls and Social Posting
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Changelog planning just got way simpler!
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Email validation, filtering, and subscriber search üîç
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Analytics Drill-Down and Localization
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Already mentioned notice
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Published Page Search and Filtering
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Improved Analytics Accessibility
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Tables, tables, tables
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Analytics Drilldown + Changelog Generator Styling
      </span>
    </div>
  </div>
</div>
`;

exports[`Test with active 1`] = `
<div
  className="Pinpoint Documentation Outline Wrapper "
>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Linked Content Blocks
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Publishing Controls and Social Posting
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item active "
        onClick={[Function]}
      >
        Changelog planning just got way simpler!
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Email validation, filtering, and subscriber search üîç
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Analytics Drill-Down and Localization
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Already mentioned notice
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Published Page Search and Filtering
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Improved Analytics Accessibility
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Tables, tables, tables
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Analytics Drilldown + Changelog Generator Styling
      </span>
    </div>
  </div>
</div>
`;

exports[`Test with click handler 1`] = `
<div
  className="Pinpoint Documentation Outline Wrapper "
>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Linked Content Blocks
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Publishing Controls and Social Posting
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Changelog planning just got way simpler!
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Email validation, filtering, and subscriber search üîç
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Analytics Drill-Down and Localization
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Already mentioned notice
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Published Page Search and Filtering
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <div
        className="spacer"
      />
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Improved Analytics Accessibility
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Tables, tables, tables
      </span>
    </div>
  </div>
  <div
    className="Parent"
  >
    <div
      className="Top"
    >
      <svg
        aria-hidden="true"
        className="toggleControl"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
        />
      </svg>
      <span
        className="Pinpoint Internal Link clickable Pinpoint Documentation Outline Item  "
        onClick={[Function]}
      >
        Analytics Drilldown + Changelog Generator Styling
      </span>
    </div>
  </div>
</div>
`;
